database/mongoDb.js

```diff
@@ -0,0 +1,214 @@
+          }
+
+        }
+      });
+
+    });
+
+    return params;
+  }
+
+
+  //function to load feature mock data
+  const loadFeatures = (callback, numRecords = 100, params = []) => {
+    console.log(documentCount);
+    if (params.length === 0) {
+      features.createFeatures(numRecords).forEach(house => {
+        params.push({
+          insertOne : {
+            document: {
+              house_id: ++documentCount,
+              type:house.type,
+              year_built: house.year_built,
+              heating: house.heating,
+              cooling: house.cooling,
+              parking: house.parking,
+              lot: house.lot,
+              days_on_zillow: house.days_on_zillow,
+              bedrooms: house.bedrooms,
+              bathrooms: house.bathrooms,
+              interiorheating: house.int_heating,
+              interiorcooling: house.int_cooling,
+              appliances: house.appliances,
+              kitchen: house.kitchen,
+              flooring: house.flooring,
+              sqft: house.sqft
+            }
+          }
+        });
+      });
+    } else {
+      params.forEach(function (value, index, array) {
+        value.insertOne.document.house_id = ++documentCount;
+      });
+    }
+
+    featuresTable.bulkWrite(params).then(()=> {
+      console.log(`Loaded ${numRecords}`);
+      callback();
+    }).catch((err)=> {
+      console.log('an error occured');
+      console.log(err);
+    });
+
+    };
+
+  /////////////////////////Retreiving Facts and Features///////////////////////////
+  const getFeatures = (id, callback) => {
+    console.log(id);
+    featuresTable.find({ house_id: id }, function (err, docs) {
+      if (err) {
+        console.log(err);
+        callback(err, null);
+      } else {
+        console.log(docs);
+        callback(null, docs);
+      }
+    });
+  };
+
+  const getBedBaths = (id, callback) => {
+    console.log(id);
+    featuresTable.find({ house_id: id }, function (err, docs) {
+      if (err) {
+        console.log(err);
+        callback(err, null);
+      } else {
+        console.log(docs);
+        callback(null, docs);
+      }
+    });
+    };
+
+    var getAllFeatures = (page, callback) => {
+
+    };
+
+    var postFeatures = (features, callback) => {
+      if (isNaN(id)) {
+        callback('no Id to insert', null);
+      }
+      features.house_id = undefined;
+      features = keepOnlyFieldNames(features);
+      featuresTable.upsert(features);
+    };
+
+    var updateFeatures = (id, features, callback) => {
+      if (isNaN(id)) {
+        callback('no Id to insert', null);
+      }
+      features.house_id = id;
+      features = keepOnlyFieldNames(features);
+    };
+
+    var deleteFeatures = (id, callback) => {
+
+    };
+
+    var keepOnlyFieldNames = ({ house_id, type, year_built, heating, cooling, parking, lot, days_on_zillow, bedrooms, bathrooms, interiorheating, interiorcooling, appliances, kitchen, flooring, sqft}) => {

```

Comment:
Why? Why is this happening?

Reply:
Wanted to make sure if the user/front-end sends in something misspelled or something random will remove these fields. Probably was overthinking it because mongoose does have a schema that would be violated and return an error rather than straight mongo which I believe can accept anything. Actually at the time thought this was a smart way only had to write the key names once then copy and paste.